company-clang.el

```diff
@@ -109,8 +179,9 @@ or automatically through a custom `company-clang-prefix-guesser'."
 
 ;; TODO: Handle Pattern (syntactic hints would be neat).
 ;; Do we ever see OVERLOAD (or OVERRIDE)?
+;; Parse completion's comments/documentation too
 (defconst company-clang--completion-pattern
-  "^COMPLETION: \\_<\\(%s[a-zA-Z0-9_:]*\\)\\(?: : \\(.*\\)$\\)?$")
+   "^COMPLETION: \\_<\\(%s[a-zA-Z0-9_:]*\\) : \\(.*?\\)\\(?: : \\(.*\\)\\)\\{,1\\}$")
```

Comment:
I wrote "impossible" clang results to test the regexp, e.g. **"COMPLETION: "** and **"COMPLETION: foobar : "** are conditions that should never be met, instead **"COMPLETION: foobar : [#int#]foobar(<#int a#>) : "** is possible (notice the empty comment) which is the 6th variant in the table above, **"COMPLETION: blabla"** and **"COMPLETION: foobar"** are still legit (if they where internals..., still have to try to add an empty Javadoc comment to an internal to get something like "COMPLETION: foobar : ").

Consider a `silly.c` file where you want to complete **foo**:
* **"COMPLETION: extern"** is a common case where the prefix could not be found, it is filtered out by `company-clang--parse-output`, but still there in \*clang-output\* buffer.
* **"COMPLETION: foobar : [#int#]foobar(<#int a#>) : "** is part of the output when clang finds an empty comment (/** */).
```C
// silly.c

// the next comment is empty: "COMPLETION: foobar : [#int#]foobar(<#int a#>) : "
/** */
int foobar(int a){return 0;}

void my_test() {
  foo
}
```